diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index c2cb1bc5b1..6e84d8ae53 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -136,7 +136,7 @@ static inline u64 unscale_slice(u64 delta, struct sched_entity *se) {
 	return __unscale_slice(delta, se->burst_score);
 }
 
-void reweight_task(struct task_struct *p, int prio);
+static void reweight_task_fair(struct rq *rq, struct task_struct *p, int prio);
 
 static void update_burst_score(struct sched_entity *se) {
 	if (!entity_is_task(se)) return;
@@ -148,7 +148,7 @@ static void update_burst_score(struct sched_entity *se) {
 
 	u8 new_prio = min(39, prio + se->burst_score);
 	if (new_prio != prev_prio)
-		reweight_task(p, new_prio);
+		reweight_task_fair(task_rq(p), p, new_prio);
 }
 
 static void update_burst_penalty(struct sched_entity *se) {
-- 
2.34.1
